### ISSUE TITLE
jcl to groovy migration fails with no jobcard

### ISSUE THREAD

**Issue by rsjrny** [IBM/dbb#151]
Is it possible to bypass this requirement? I have hundreds of PROCS to migrate and none have JOB cards since they are PROCS.

```
DMH1471I Error found in  at line 0                      
DMH1334E First statement was not JOB card               
DMH1375E Missing JOB card        
```

**Comment by dennis-behm** [comment]
Hi @rsjrny ,

The messages which you see are resulting from the included scanner shipped with DBB. It is a JCL scanner - you see it at

https://github.com/IBM/dbb/blob/b979a6d73ab8756c64e1461dfd37d32bce0badbf/Migration/jcl/groovy/JCLtoDBB.groovy#L127-L128 It is not in the control of the script itself. 

I am wondering if your approach makes sense to convert all your procs 1:1 to groovy scripts. I have to admin, I have too little insights.

Typical, we recommend users, to perform an assessment to map existing functions to the new ways of building the application (based on the zAppBuild sample). Hopefully this will reduce the number of procs which need to be migrated.

It is a matter of the design your are taking to implement DBB and the pipeline. An alternative could be to leverage the JCLExec API in your new build frame to reuse PROCs ( when it makes sense ). 

Dennis

**Comment by rsjrny** [comment]
I have to agree that it does not make sense to convert all of them but here is the issue. I will use my HLASM as an example

I have the following procs for example (a simple example)

hlasm01 - NORENT
hlasm02 - RENT
hlasm03 - AC=1

Then for each of those procs I have overrides for different levels of maclibs and other things, I have to support different operating systems

asm01 uses hlasm01 mac1.2
asm02 uses hlasm01 mac 1.3
asm03 uses hlasm02 mac 1.2
asm04 uses hlasm02 mac 1.3
asm05 uses hlasm03 mac 1.2
asm06 uses hlasm03 mac 1.3

In order to use the application-conf in zAppBuild

I need to create an asm01.groovy and an asm01.properties which looks like my hlasm01 proc with asm01 overrides
I need to create an asm02.groovy and an asm03.properties which looks like my hlasm01 proc with asm02 overrides
I need to create an asm03.groovy and an asm03.properties which looks like my hlasm01 proc with asm03 overrides
etc. for each hlasm proc

The simple solution would be if I could write am HLASM01.groovy and have multiple HLASMnn.properties files that are assigned to the same groovy script. for example, 

if my .ext was asm01 it would use HLASM01.groovy and asm01.properties,
if my .ext was asm02 it would use HLASM01.groovy and asm02.properties, 
if my .ext was asm03 it would use HLASM01.groovy and asm03.properties

I do not see a way to do that in zappbuild so I think I have to create a groovy script for each PROC and PROC overrides.

what do you think @dennis-behm 

thanks,
